lib/Runtime/Base/FunctionExecutionStateMachine.cpp

```diff
@@ -50,10 +50,16 @@ namespace Js
 
         const ConfigFlagsTable &configFlags = Configuration::Global.flags;
 
+        // AutoProfilingInterpreter might decide to not profile on the first run. For generator
+        // functions, that means we will miss the profiling information on the first run when we resume
+        // back to the function. This might result in lots of BailOnNoProfile even though we should have
+        // profiling information. So always collect profile data for generators
+        const bool skipAutoProfileForGenerator = functionBody->SkipAutoProfileForCoroutine();
+
         interpreterLimit = 0;
-        autoProfilingInterpreter0Limit = static_cast<uint16>(configFlags.AutoProfilingInterpreter0Limit);
+        autoProfilingInterpreter0Limit = skipAutoProfileForGenerator ? 0 : static_cast<uint16>(configFlags.AutoProfilingInterpreter0Limit);
         profilingInterpreter0Limit = static_cast<uint16>(configFlags.ProfilingInterpreter0Limit);
```

Comment:
in case we are disabling autoProfiling here, we probably want `profilingInterpreter0Limit` to be `configFlags.AutoProfilingInterpreter0Limit + configFlags.AutoProfilingInterpreter1Limit` so we don't JIT too aggressively